flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/TableEnvironmentTest.scala

```diff
@@ -18,231 +18,574 @@
+        util.addTable[CClass]('cf1, 'cf3 as 'new, 'cf2),
+        Seq("cf1" -> INT, "new" -> DOUBLE, "cf2" -> STRING))
+
+      // row
+      util.verifySchema(
+        util.addTable('rf1, 'rf3 as 'new, 'rf2)(TEST_ROW),
+        Seq("rf1" -> INT, "new" -> DOUBLE, "rf2" -> STRING))
+
+      // tuple
+      util.verifySchema(
+        util.addTable[JTuple3[Int, String, Double]]('f0, 'f2 as 'new, 'f1),
+        Seq("f0" -> INT, "new" -> DOUBLE, "f1" -> STRING))
+    }
   }
 
   @Test
-  def testGetFieldInfoCClassNames(): Unit = {
-    val fieldInfo = tEnv.getFieldInfo(
-      caseClassType,
-      Array(
-        UnresolvedFieldReference("name1"),
-        UnresolvedFieldReference("name2"),
-        UnresolvedFieldReference("name3")
-      ))
-
-    fieldInfo._1.zip(Array("name1", "name2", "name3")).foreach(x => assertEquals(x._2, x._1))
-    fieldInfo._2.zip(Array(0, 1, 2)).foreach(x => assertEquals(x._2, x._1))
+  def testStreamAliasWithAddingTimeAttributesByName(): Unit = {
+    val util = streamTestUtil()
+
+    // atomic
+    util.verifySchema(
+      util.addTable[Int](('myint as 'new).proctime),
+      Seq("new" -> PROCTIME))
+
+    // case class
+    util.verifySchema(
+      util.addTable[CClassWithTime]('cf1, ('newnew as 'new).proctime, 'cf2),
+      Seq("cf1" -> INT, "new" -> PROCTIME, "cf2" -> LONG))
+
+    util.verifySchema(
+      util.addTable[CClassWithTime]('cf1, ('newnew as 'new).rowtime, 'cf2),
+      Seq("cf1" -> INT, "new" -> ROWTIME, "cf2" -> LONG))
+
+    // row
+    util.verifySchema(
+      util.addTable('rf1, ('newnew as 'new).proctime, 'rf2)(TEST_ROW_WITH_TIME),
+      Seq("rf1" -> INT, "new" -> PROCTIME, "rf2" -> LONG))
+
+    util.verifySchema(
+      util.addTable('rf1, ('newnew as 'new).rowtime, 'rf2)(TEST_ROW_WITH_TIME),
+      Seq("rf1" -> INT, "new" -> ROWTIME, "rf2" -> LONG))
+
+    // tuple
+    util.verifySchema(
+      util.addTable[JTuple3[Int, Long, String]]('f0, ('newnew as 'new).proctime, 'f1),
+      Seq("f0" -> INT, "new" -> PROCTIME, "f1" -> LONG))
+
+    util.verifySchema(
+      util.addTable[JTuple3[Int, Long, String]]('f0, ('newnew as 'new).rowtime, 'f1),
+      Seq("f0" -> INT, "new" -> ROWTIME, "f1" -> LONG))
   }
 
   @Test
-  def testGetFieldInfoPojoNames2(): Unit = {
-    val fieldInfo = tEnv.getFieldInfo(
-      pojoType,
-      Array(
-        UnresolvedFieldReference("pf3"),
-        UnresolvedFieldReference("pf1"),
-        UnresolvedFieldReference("pf2")
-      ))
-
-    fieldInfo._1.zip(Array("pf3", "pf1", "pf2")).foreach(x => assertEquals(x._2, x._1))
-    fieldInfo._2.zip(Array(2, 0, 1)).foreach(x => assertEquals(x._2, x._1))
+  def testStreamAliasWithReplacingTimeAttributesByName(): Unit = {
+    val util = streamTestUtil()
+
+    // atomic
+    util.verifySchema(
+      util.addTable[Int](('myint as 'new).proctime),
+      Seq("new" -> PROCTIME))
+
+    // case class
+    util.verifySchema(
+      util.addTable[CClassWithTime]('cf1, ('cf3 as 'new).proctime, 'cf2),
```

Comment:
Should we allow to define an existing field as a `proctime` attribute?

Reply:
I removed this.